server/data/projects.js

```diff
@@ -94,6 +103,65 @@ const createProject = async (projectObjParam, user) => {
 	return createdProject;
 };
 
+const updateProject = async (projectObjParam, id, user) => {
+	id = isValidObjectId(id);
+	let projectCheck = await getProjectById(id);
+	ownercheck = checkuseraccess(user, projectCheck.owner);
+	if (!ownercheck)
+		throw forbiddenErr(
+			`Not Authorised to update this project. Not Project Owner`
+		);
+	const projectObj = isValidProjectObject(projectObjParam);
+	let { name, description, github, media, technologies, deploymentLink } =
+		projectObj;
+		name = isValidProjectName(name);
+		description = description
+			? isValidStr(description, 'project description')
+			: null;
+		github = github ? isValidGithub(github) : null;
+		technologies = isValidTechnologies(technologies);
+		deploymentLink = deploymentLink
+			? isValidStr(deploymentLink, 'project deployment link')
+			: null;
+	const date = new Date();
+	t = updateProject;
+	const updateProjectObject = {
+		...updateProject,
+		name,
+		description,
+		github,
+		media,
+		deploymentLink,
+		updatedAt: date,
+		technologies,
+	};
+	const projectCollection = await projects();
+	const updateInfo = await projectCollection.updateOne(
+		{ _id: ObjectId(id) },
+		{ $set: updateProjectObject }
+	);
+	const project = await getProjectById(id);
+	if (!project) throw notFoundErr('No Project found ');
```

Comment:
getProjectById will throw a not found error if there is no project found, you don't have to do it here